### PR TITLE
--show-snippets with default true and off in tests

### PR DESCRIPTION
**Pull request by davidcok**
Adjusts the tests so that if --show-snippets is on by default, the tests do not fail.
Also sets --show-snippets on by default.

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

**Comment by keyboardDrummer** [comment]
I'm OK with this change, but requested some others to review as well.

File: docs/check-examples [hunk 1]
```diff
@@ -379,11 +380,11 @@ do
         else
           if [ "$command" == "%check-legacy" ]; then
             isverbose=-compileVerbose:$checkIds
-            (cd "$dir"; "$dafny" $verb $dOptions $isverbose ${options[@]} $text   ) > "$actualOut" 2> "$actualError"
+            (cd "$dir"; "$dafny" $verb /showSnippets:0 $dOptions $isverbose ${options[@]} $text   ) > "$actualOut" 2> "$actualError"
```
Comment (MikaelMayer) [inline docs/check-examples:383]: @keyboardDrummer  This looks suspicious, it should probably be --show-snippets:false

Comment (davidcok) [reply docs/check-examples:383]: $verb is always empty at this point -- this is the branch for legacy CLI tests. SO it is correct, but perhaps $verb could be removed.

Comment (davidcok) [reply docs/check-examples]: Removed $verb on this line to avoid confusion.

Comment (davidcok) [reply docs/check-examples]: And tests now pass after nightlies have been fixed.  Mikael - can you approve now?

**Review by robin-aws — COMMENTED** [review]
I'm fine with changing the default here since most users aren't going to be parsing console output, and even if they are broken by this it's just a matter of adding `--show-snippets:false`.

But on that note, do add a release note to explicitly call this change out, so it's easier for users to spot it.

(not explicitly requesting changes so another reviewer is free to approve once this is done)

**Comment by davidcok** [comment]
Mikael (or Robin) - I have added a release note per Robin’s request. Please approve the PR or let me know what else needs improvement.

Thanks - David

**Review by keyboardDrummer — APPROVED** [review]

**Pull request merged**